common/weave/client.go

```diff
@@ -27,29 +27,47 @@ type Client interface {
 type Status struct {
 	Version string
 	Router  Router
-	DNS     DNS
-	IPAM    IPAM
+	DNS     *DNS
+	IPAM    *IPAM
 }
 
 // Router describes the status of the Weave Router
 type Router struct {
-	Name  string
-	Peers []struct {
+	Name               string
+	Encryption         bool
+	ProtocolMinVersion int
+	ProtocolMaxVersion int
+	PeerDiscovery      bool
+	Peers              []Peer
+	Connections        []struct {
+		Address  string
+		Outbound bool
+		State    string
+		Info     string
+	}
+	Targets        []string
+	TrustedSubnets []string
+}
```

Comment:
and since we couldn't vendor the whole data structure (that would imply vendoring `weave/prog/weaver/http.go` which is part of a program not a library) I decided to continue replicate the parts I needed.